### PR TITLE
Add filter tags to samples page

### PR DESCRIPTION
**Pull request by cdebourcy**

### CODE REVIEW

**Comment by cdebourcy** [comment]
screen shot 2018-03-14 at 10 28 08 am

File: app/assets/src/components/Samples.jsx [hunk 1]
```diff
@@ -917,7 +917,42 @@ class Samples extends React.Component {
     location.href = `/projects/${id}/csv`;
   }
 
+  applyExcluded(e, type, state_var) {
+    let id = e.target.getAttribute('data-exclude')
+    if (type === "int") {
+      id = +id
+    }
+    let list = this.state[state_var].splice(0);
```
Comment (yunfangjuan) [inline app/assets/src/components/Samples.jsx]: use Object.assign instead if you intend to copy the content?

File: app/assets/src/components/Samples.jsx [hunk 2]
```diff
@@ -917,7 +917,42 @@ class Samples extends React.Component {
...
+    let index;
+    index = list.indexOf(id);
+    list.splice(index, 1);
```
Comment (yunfangjuan) [inline app/assets/src/components/Samples.jsx]: check index >= 0 in case of contention

File: app/assets/src/components/Samples.jsx (hunk 1 above)
Comment (cdebourcy) [reply app/assets/src/components/Samples.jsx]: ok

File: app/assets/src/components/Samples.jsx (hunk 2 above)
Comment (cdebourcy) [reply app/assets/src/components/Samples.jsx]: ok

**Pull request merged**